src/main.ts

```diff
@@ -156,33 +185,57 @@ ArgoCD Diff for commit [\`${shortCommitSha}\`](${commitLink})
   }
 }
 
+async function asyncForEach<T>(
+  array: T[],
+  callback: (item: T, i: number, arr: T[]) => Promise<void>
+): Promise<void> {
+  for (let index = 0; index < array.length; index++) {
+    await callback(array[index], index, array);
+  }
+}
+
 async function run(): Promise<void> {
   const argocd = await setupArgoCDCommand();
   const apps = await getApps();
   core.info(`Found apps: ${apps.map(a => a.metadata.name).join(', ')}`);
 
-  const diffPromises = apps.map(async app => {
+  const diffs: Diff[] = [];
+
+  await asyncForEach(apps, async app => {
```

Comment:
Argo isn't good at handling parallel invocations?

Reply:
🦅 👁️ 
The server can handle it, but the CLI can't. Apparently the `argocd app diff` command creates temp folders for the helm diff output, and running things concurrently was throwing errors about the `tmp` folder being unwritable.